File: frontend/src/types/announcement.ts
export interface AnnouncementDTO {
  id: string
  course_id: string
  org_id: string
  author_id: string
  title: string
  content: string
  created_at: string
  updated_at: string
}

export interface CreateAnnouncementInput {
  title: string
  content: string
}

File: backend/internal/domain/entities/announcement.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Announcement struct {
	ID        uuid.UUID
	CourseID  uuid.UUID
	OrgID     uuid.UUID
	AuthorID  uuid.UUID
	Title     string
	Content   string
	CreatedAt time.Time
	UpdatedAt time.Time
}
